src/datepicker/datepicker.js

```diff
@@ -543,19 +547,21 @@ angular.module('ui.bootstrap.datepicker', ['ui.bootstrap.dateparser', 'ui.bootst
   altInputFormats: []
 })
 
-.controller('UibDatepickerPopupController', ['$scope', '$element', '$attrs', '$compile', '$parse', '$document', '$rootScope', '$uibPosition', 'dateFilter', 'uibDateParser', 'uibDatepickerPopupConfig', '$timeout',
-function(scope, element, attrs, $compile, $parse, $document, $rootScope, $position, dateFilter, dateParser, datepickerPopupConfig, $timeout) {
+.controller('UibDatepickerPopupController', ['$scope', '$element', '$attrs', '$compile', '$parse', '$document', '$rootScope', '$uibPosition', 'dateFilter', 'uibDateParser', 'uibDatepickerPopupConfig', '$timeout', '$log',
+function(scope, element, attrs, $compile, $parse, $document, $rootScope, $position, dateFilter, dateParser, datepickerPopupConfig, $timeout, $log) {
   var self = this;
   var cache = {},
     isHtml5DateInput = false;
   var dateFormat, closeOnDateSelection, appendToBody, onOpenFocus,
     datepickerPopupTemplateUrl, datepickerTemplateUrl, popupEl, datepickerEl,
-    ngModel, $popup, altInputFormats;
+    ngModel, ngModelOptions, $popup, altInputFormats;
 
   scope.watchData = {};
 
-  this.init = function(_ngModel_) {
+  this.init = function(_ngModel_, _ngModelOptions_) {
     ngModel = _ngModel_;
+
+    ngModelOptions = _ngModelOptions_ && _ngModelOptions_.$options || {};
```

Comment:
@jbarrus @wesleycho is right. All that ngModelOption controller hook-up seems unnecessary; my bad. We can just look at `ngModel.$options`. I plan to update my wip soon.